services/core/IEEE_2030_5/ieee_2030_5/agent.py

```diff
@@ -0,0 +1,804 @@
+        resp = self._client.put_der_settings(href, new_settings)
+        return resp
+
+    @RPC.export
+    def update_der_availability(self, href: str, new_availability: m.DERAvailability) -> int:
+        resp = self._client.put_der_availability(href, new_availability)
+        return resp
+
+    @RPC.export
+    def update_der_status(self, href: str, new_availability: m.DERAvailability) -> int:
+        resp = self._client.put_der_status(href, new_availability)
+        return resp
+
+    @RPC.export
+    def get_der_references(self) -> List[str]:
+        return self._client.get_der_hrefs()
+
+    def _active_controls_changed(self, active: m.DERControlList):
+        if not isinstance(active, m.DERControlList):
+            _log.error("Invalid instance passed to active control changed")
+            return
+        _log.debug(active)
+
+    def _default_control_changed(self, default_control: m.DefaultDERControl):
+        if not isinstance(default_control, m.DefaultDERControl):
+            _log.error("Invalid instance of default control")
+            raise ValueError(f"Invalid instance of default control was {type(default_control)}")
+
+        if self._current_control is not None:
+            _log.info("Default config has been overwritten by event.")
+            return
+        _log.info("Sending controls to platform.driver")
+
+        self._default_der_control = default_control
+
+        default_control_points = list(
+            filter(lambda x: 'DefaultDERControl' in x.parameter_type,
+                   self._mapped_points.values()))
+        der_base_points = list(
+            filter(lambda x: 'DERControlBase' in x.parameter_type, self._mapped_points.values()))
+
+        for point in default_control_points:
+            point_value = getattr(default_control, point.parameter)
+
+            try:
+                if point_value:
+                    if not isinstance(point_value, (float, int, bool)):
+                        point_value = getattr(point_value, "value")
+
+                    if point_value:
+                        self.vip.rpc.call("platform.driver", "set_point", self._device_topic,
+                                          point.point_on_bus, point_value)
+            except TypeError:
+                _log.error(f"Error setting point {point.point_on_bus} to {point_value}")
+            except KeyError:
+                _log.error(f"Error setting point {point.point_on_bus} to {point_value}")
+
+        for point in der_base_points:
+
+            point_value = getattr(default_control.DERControlBase, point.parameter)
+
+            try:
+                if point_value:
+                    if not isinstance(point_value, (float, int, bool)):
+                        point_value = getattr(point_value, "value")
+
+                    if point_value:
+                        self.vip.rpc.call("platform.driver", "set_point", point.point_on_bus,
```

Comment:
set_point requires device path, point name and value

Reply:
Fixed